llama_index/evaluation/base.py

```diff
@@ -1,16 +1,38 @@
 """Evaluating the responses from an index."""
 from __future__ import annotations
 
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
 from typing import List, Optional
 
-from llama_index import (
-    Document,
-    GPTListIndex,
-    QuestionAnswerPrompt,
-    RefinePrompt,
-    Response,
-    ServiceContext,
-)
+from llama_index.indices.base import ServiceContext
+from llama_index.indices.list.base import GPTListIndex
+from llama_index.prompts.prompts import QuestionAnswerPrompt, RefinePrompt
+from llama_index.readers.schema.base import Document
+from llama_index.response.schema import Response
+
+
+@dataclass
+class Evaluation:
+    passing: bool = False  # True if the response is correct, False otherwise
+    feedback: str = ""  # Feedback for the response
+
+
+class BaseEvaluator(ABC):
+    def __init__(self, service_context: Optional[ServiceContext] = None) -> None:
+        """Base class for evaluating responses"""
+        self.service_context = service_context or ServiceContext.from_defaults()
+
+    @abstractmethod
+    def evaluate_response(self, query: str, response: Response) -> Evaluation:
+        """Evaluate the response for a query and return an Evaluation."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def evaluate(self, query: str, response: Response) -> str:
```

Comment:
could we 1) deprecate this function, and 2) just have this return evaluate_response by default? e.g. 
```
eval_result = self.evaluate_response(query, response)
return "YES" if eval_result.passing else "NO"
```